File: Eye-openingJava/src/main/java/com/example/eye_openingJava/ui/common/view/TypefaceTextView.java
package com.example.eye_openingJava.ui.common.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

import com.example.eye_openingJava.R;
import com.example.eye_openingJava.util.TypeFaceUtil;

/**
 *@Author: create by boge
 *@Createtime: 2023/9/6 16:26
 *@Function:
 * 带有自定义字体TextView。
*/
public class TypefaceTextView extends AppCompatTextView {

    private static Context mContext;

    public TypefaceTextView(Context context) {
        this(context, null);
        mContext = context;
    }

    public TypefaceTextView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
        mContext = context;
    }

    public TypefaceTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        if (attrs != null) {
            TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.TypefaceTextView, 0, 0);
            int typefaceType = typedArray.getInt(R.styleable.TypefaceTextView_typeface, 0);
            setTypeface(getTypeface(typefaceType));
            typedArray.recycle();
        }

    }

    /**
     * 根据字体类型，获取自定义字体。
     */
    public static Typeface getTypeface(Integer typefaceType) {
        switch (typefaceType) {
            case TypeFaceUtil.FZLL_TYPEFACE:
                return TypeFaceUtil.getFontTypeface(mContext,TypeFaceUtil.FZLL_TYPEFACE);
            case TypeFaceUtil.FZDB1_TYPEFACE:
                return TypeFaceUtil.getFontTypeface(mContext,TypeFaceUtil.FZDB1_TYPEFACE);
            case TypeFaceUtil.FUTURA_TYPEFACE:
                return TypeFaceUtil.getFontTypeface(mContext,TypeFaceUtil.FUTURA_TYPEFACE);
            case TypeFaceUtil.DIN_TYPEFACE:
                return TypeFaceUtil.getFontTypeface(mContext,TypeFaceUtil.DIN_TYPEFACE);
            case TypeFaceUtil.LOBSTER_TYPEFACE:
                return TypeFaceUtil.getFontTypeface(mContext,TypeFaceUtil.LOBSTER_TYPEFACE);
            default:
                return Typeface.DEFAULT;
        }
    }
}

File: app/src/main/java/com/example/eyeOpeningKotlin/ui/common/view/TypefaceTextView.kt
package com.example.eyeOpeningKotlin.ui.common.view

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.example.eyeOpeningKotlin.R
import com.example.eyeOpeningKotlin.util.TypeFaceUtil

/**
 * 带有自定义字体TextView。
 *
 * @author driverSkr
 * @since  2023/10/18
 */
class TypefaceTextView: AppCompatTextView {

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        attrs?.let {
            val typedArray = context.obtainStyledAttributes(it, R.styleable.TypefaceTextView, 0, 0)
            val typefaceType = typedArray.getInt(R.styleable.TypefaceTextView_typeface, 0)
            typeface = getTypeface(typefaceType)
            typedArray.recycle()
        }
    }

    companion object{
        /**
         * 根据字体类型，获取自定义字体。
         */
        fun getTypeface(typefaceType: Int?): Typeface = when(typefaceType){
            TypeFaceUtil.FZLL_TYPEFACE -> TypeFaceUtil.fzlLTypeface
            TypeFaceUtil.FZDB1_TYPEFACE -> TypeFaceUtil.fzdb1Typeface
            TypeFaceUtil.FUTURA_TYPEFACE -> TypeFaceUtil.futuraTypeface
            TypeFaceUtil.DIN_TYPEFACE -> TypeFaceUtil.dinTypeface
            TypeFaceUtil.LOBSTER_TYPEFACE -> TypeFaceUtil.lobsterTypeface
            else -> Typeface.DEFAULT
        }
    }
}